src/lib/components/store/Store.js

```diff
@@ -0,0 +1,101 @@
+import React, { Component } from 'react';
+
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import DesignCollection from '../design_collection/DesignCollection';
+import CartButton from '../cart_button/CartButton';
+import Pagination from '../pagination/Pagination';
+import StoreFilter from '../store_filter/StoreFilter';
+
+import './Store.css';
+
+const CLASS_ROOT = 'tp-store';
+
+export default class Store extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      page: props.page || 1,
+      albumId: parseInt(props.albumId, 10),
+      productTypeName: props.productTypeName
+    };
+  }
+
+  pageChangeHandler = page => {
+    this.setState({ page: page });
+    const { albumId, productTypeName } = this.state;
+    this.fetchStore(page, albumId, productTypeName);
+  };
+
+  albumChangeHandler = albumId => {
+    this.setState({ albumId: albumId });
+    const { page, productTypeName } = this.state;
+    this.fetchStore(page, albumId, productTypeName);
+  };
+
+  productTypeChangeHandler = productTypeName => {
+    this.setState({ productTypeName: productTypeName });
+    const { page, albumId } = this.state;
+    this.fetchStore(page, albumId, productTypeName);
+  };
+
+  fetchStore = (page, albumId, productTypeName) => {
+    const { storeUrl } = this.props.configuration.storeUrl;
```

Comment:
@jeffcooper86  @nitishgargiitd Should this `storeUrl` actually be a function? That way you can pass through what you want to do -- GET request to store, then set state? Or us a URL ok for now?

Reply:
A function makes sense to me.

Reply:
Yeah makes sense to me. Integration layer will need to have control over the storeUrl building logic, which makes sure that and component user can build proper URL according to their backend